From 598122a967ea9301a8e96d95e50e567f07d0099c Mon Sep 17 00:00:00 2001
From: Fabrice Goucem <fabrice.goucem@nxp.com>
Date: Mon, 25 Oct 2021 17:18:31 +0200
Subject: [PATCH 1/3] arm64: dts: imx8mm-evk-harpoon, imx8mn-evk-harpoon,
 imx8mp-evk-harpoon: new DTBs for real-time-edge

Signed-off-by: Fabrice Goucem <fabrice.goucem@nxp.com>
[replace spaces by tabulations]
Signed-off-by: Stephane Viau <stephane.viau@nxp.com>
(cherry picked from commit 82b9700c486433119fcf93632157c5ed587f1dc0)
---
 arch/arm64/boot/dts/freescale/Makefile            |  6 +++---
 .../boot/dts/freescale/imx8mm-evk-harpoon.dts     | 15 +++++++++++++++
 .../boot/dts/freescale/imx8mn-evk-harpoon.dts     | 15 +++++++++++++++
 .../boot/dts/freescale/imx8mp-evk-harpoon.dts     | 15 +++++++++++++++
 4 files changed, 48 insertions(+), 3 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mn-evk-harpoon.dts
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-evk-harpoon.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 1501d3d50168..1096ff1aebef 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -80,7 +80,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx8mm-evk.dtb imx8mm-evk-rpmsg.dtb imx8mm-evk-rm67191
 			  imx8mm-evk-qca-wifi.dtb \
 			  imx8mm-evk-rm67199.dtb \
 			  imx8mm-evk-dpdk.dtb \
-			  imx8mm-evk-lk.dtb
+			  imx8mm-evk-lk.dtb imx8mm-evk-harpoon.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mm-evk-rpmsg-wm8524.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mm-evk-ak4497.dtb imx8mm-evk-ak5558.dtb imx8mm-evk-audio-tdm.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mm-evk-8mic-revE.dtb imx8mm-evk-8mic-swpdm.dtb \
@@ -102,13 +102,13 @@ dtb-$(CONFIG_ARCH_MXC) += imx8mn-evk.dtb imx8mn-evk-rm67191.dtb imx8mn-evk-rpmsg
 			  imx8mn-ddr3l-evk-ak5558.dtb imx8mn-ddr3l-evk-rpmsg.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mn-ddr4-evk.dtb imx8mn-ddr4-evk-ak5558.dtb imx8mn-ddr4-evk-rm67191.dtb \
 			  imx8mn-ddr4-evk-rpmsg.dtb imx8mn-ddr4-evk-usd-wifi.dtb imx8mn-ddr4-evk-rm67199.dtb
-dtb-$(CONFIG_ARCH_MXC) += imx8mn-ddr4-evk-root.dtb imx8mn-ddr4-evk-inmate.dtb imx8mn-evk-root.dtb imx8mn-evk-inmate.dtb imx8mn-evk-lk.dtb imx8mn-ddr4-evk-lk.dtb
+dtb-$(CONFIG_ARCH_MXC) += imx8mn-ddr4-evk-root.dtb imx8mn-ddr4-evk-inmate.dtb imx8mn-evk-root.dtb imx8mn-evk-inmate.dtb imx8mn-evk-lk.dtb imx8mn-ddr4-evk-lk.dtb imx8mn-evk-harpoon.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mn-ab2.dtb imx8mn-ddr4-ab2.dtb imx8mn-ddr3l-ab2.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mn-var-som-symphony.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-evk.dtb imx8mp-evk-rm67191.dtb imx8mp-evk-it6263-lvds-dual-channel.dtb \
 			  imx8mp-evk-pcie-ep.dtb  imx8mp-evk-rpmsg.dtb imx8mp-evk-ecspi-slave.dtb \
 			  imx8mp-evk-jdi-wuxga-lvds-panel.dtb imx8mp-evk-flexcan2.dtb \
-			  imx8mp-evk-root.dtb imx8mp-evk-inmate.dtb imx8mp-evk-ov2775.dtb \
+			  imx8mp-evk-root.dtb imx8mp-evk-harpoon.dtb imx8mp-evk-inmate.dtb imx8mp-evk-ov2775.dtb \
 			  imx8mp-evk-ov2775-ov5640.dtb imx8mp-evk-basler-ov5640.dtb imx8mp-evk-basler.dtb \
 			  imx8mp-evk-basler-ov2775.dtb imx8mp-evk-dual-basler.dtb \
 			  imx8mp-evk-dual-ov2775.dtb imx8mp-evk-spdif-lb.dtb \
diff --git a/arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts b/arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts
new file mode 100644
index 000000000000..47e16e606332
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2021 NXP
+ */
+
+#include "imx8mm-evk-root.dts"
+
+&i2c3 {
+       status = "disabled";
+};
+
+&iomuxc {
+       pinctrl-names = "default";
+       pinctrl-0 = <&pinctrl_i2c3>;
+};
diff --git a/arch/arm64/boot/dts/freescale/imx8mn-evk-harpoon.dts b/arch/arm64/boot/dts/freescale/imx8mn-evk-harpoon.dts
new file mode 100644
index 000000000000..2700440df445
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mn-evk-harpoon.dts
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2021 NXP
+ */
+
+#include "imx8mn-evk-root.dts"
+
+&i2c3 {
+       status = "disabled";
+};
+
+&iomuxc {
+       pinctrl-names = "default";
+       pinctrl-0 = <&pinctrl_i2c3>;
+};
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-evk-harpoon.dts b/arch/arm64/boot/dts/freescale/imx8mp-evk-harpoon.dts
new file mode 100644
index 000000000000..cc153b97339e
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-evk-harpoon.dts
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2021 NXP
+ */
+
+#include "imx8mp-evk-root.dts"
+
+&i2c3 {
+	status = "disabled";
+};
+
+&iomuxc {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c3>;
+};
-- 
2.34.1


From e2b5c6f7b47f37ec568fb51bee7275038b919ace Mon Sep 17 00:00:00 2001
From: Fabrice Goucem <fabrice.goucem@nxp.com>
Date: Tue, 23 Nov 2021 19:56:55 +0100
Subject: [PATCH 2/3] arm64: dts: imx8mm-evk-harpoon: remove sai5

sai5 is used by Harpoon's RTOS:
* delete it from Linux device tree
* handle sai5 iomux through iomuxc node

Signed-off-by: Fabrice Goucem <fabrice.goucem@nxp.com>
[remove JIRA ID]
Signed-off-by: Stephane Viau <stephane.viau@nxp.com>
(cherry picked from commit f1803bc0753dc6616a1fbf60387736e05bf500be)
---
 .../boot/dts/freescale/imx8mm-evk-harpoon.dts  | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts b/arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts
index 47e16e606332..7b6f678e831c 100644
--- a/arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts
@@ -6,10 +6,22 @@
 #include "imx8mm-evk-root.dts"
 
 &i2c3 {
-       status = "disabled";
+	status = "disabled";
 };
 
 &iomuxc {
-       pinctrl-names = "default";
-       pinctrl-0 = <&pinctrl_i2c3>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c3>, <&pinctrl_sai5>;
+	pinctrl_sai5: sai5grp {
+		fsl,pins = <
+			MX8MM_IOMUXC_SAI5_RXD1_SAI5_TX_SYNC	0xd6
+			MX8MM_IOMUXC_SAI5_RXD2_SAI5_TX_BCLK	0xd6
+			MX8MM_IOMUXC_SAI5_RXD3_SAI5_TX_DATA0	0xd6
+			MX8MM_IOMUXC_SAI5_RXD0_SAI5_RX_DATA0	0xd6
+		>;
+	};
+};
+
+&sai5 {
+	status = "disabled";
 };
-- 
2.34.1


From 8ef6f5cb1768f381305b278a5720378505143127 Mon Sep 17 00:00:00 2001
From: Jiafei Pan <Jiafei.Pan@nxp.com>
Date: Mon, 13 Dec 2021 10:07:53 +0800
Subject: [PATCH 3/3] arm64: dts: imx8mp-evk-harpoon: Enable SAI audio case in
 inmate cell

1. Set pinmux in root cell.
2. Disabled SAI in root cell.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>
[Move changes from -root to -harpoon dts]
Signed-off-by: Stephane Viau <stephane.viau@nxp.com>
(cherry picked from commit 67e6b53ef9137c97aa369d1dc93e26b18a505f5a)
---
 .../boot/dts/freescale/imx8mp-evk-harpoon.dts | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-evk-harpoon.dts b/arch/arm64/boot/dts/freescale/imx8mp-evk-harpoon.dts
index cc153b97339e..1fe8004a84f8 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-evk-harpoon.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-evk-harpoon.dts
@@ -11,5 +11,22 @@ &i2c3 {
 
 &iomuxc {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c3>;
+	pinctrl-0 = <&pinctrl_i2c3>, <&pinctrl_sai3>, <&pinctrl_sai5>;
+
+	pinctrl_sai5: sai5grp {
+		fsl,pins = <
+			MX8MP_IOMUXC_SAI5_RXD1__AUDIOMIX_SAI5_TX_SYNC	0xd6
+			MX8MP_IOMUXC_SAI5_RXD2__AUDIOMIX_SAI5_TX_BCLK	0xd6
+			MX8MP_IOMUXC_SAI5_RXD3__AUDIOMIX_SAI5_TX_DATA00	0xd6
+			MX8MP_IOMUXC_SAI5_RXD0__AUDIOMIX_SAI5_RX_DATA00	0xd6
+		>;
+	};
+};
+
+&sai3 {
+	status = "disabled";
+};
+
+&sai5 {
+	status = "disabled";
 };
-- 
2.34.1

